reference/Upload-API.yaml

```diff
@@ -879,5 +879,299 @@ paths:
                       etag: 55a64df551425fcc55e4d42a148795d9f25f89d6
         description: The request body should contain a reference to each part along with the Etag of each upload.
     parameters: []
+  '/api/v1/{apiUserId}/{apiKey}/{targetId}/{languageCode}/search':
+    parameters:
+      - schema:
+          type: string
+        name: apiUserId
+        in: path
+        required: true
+        description: Guid identifying the requesting client
+      - schema:
+          type: string
+        name: apiKey
+        in: path
+        required: true
+        description: Second layer encrypted identifier for the requesting client
+      - schema:
+          type: string
+        name: targetId
```

Comment:
This is a minor problem...
OpenAPISpec **does not** allow optional path param.

An alternative I can think is having 2 paths:
1. /api/v1/{apiUserId}/{apiKey}/{targetId}/{languageCode}/search
2. /api/v1/{apiUserId}/{apiKey}/{languageCode}/search

Reply:
Will go with option 1

Reply:
Have made targetId compulsory and making another endpoint for unrestricted search that is without target id

Reply:
Another endpoint added for "Unrestricted article search"